File: Python/2021/1stOct_Rishabh-792.py
# -*- coding: utf-8 -*-
"""
Created on Fri Oct  1 18:21:03 2021

@author: Rishabh
"""
import math

def isPrime(num):
    if(num <= 1):
        return False
    x = num ** (0.5)
    x = math.ceil(x)
    
    for i in range(2,x):
        if(num % i == 0):
            return False
    return True


n = int(input("Enter no. of terms : "))
arr = []
arr.append(1)
arr.append(1)

print(arr[0])
print(arr[1])

for i in range(2,n):
    x = arr[i-1] + arr[i-2]
    arr.append(x)
    
    if( (isPrime(arr[i]))  or (arr[i] % 5 == 0)):
        print(0)
    else:
        print(arr[i])

File: JAVA/2021/1stOct_Rishabh-792.java
import java.util.Scanner;

public class Main {

    public static boolean isPrime (int num){
        for(int i=2; i*i <= num; i++){
            if(num % i == 0){
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter n - ");  
        int n = sc.nextInt();
        int [] arr;
        arr = new int [n]; 
        arr[0] = 1;
        arr[1] = 1;

        System.out.print(arr[0] + " " + arr[1] + " ");

        for(int i = 2 ; i < n ; i++){
            arr[i] = arr[i-1] + arr[i-2];

            if(isPrime(arr[i]) || (arr[i]%5 == 0)){
                System.out.print(0 + " ");
            }
            else{
                System.out.print(arr[i] + " ");
            }
        }
    }
}
